rules/rules-reviewed/eap8/eap6/eap8-xml-binding.windup.xml

```diff
@@ -0,0 +1,161 @@
+                    <location>METHOD_CALL</location>
+                </javaclass>
+                
+            </when>
+            <perform>
+                <iteration over="instanceMethods">
+                    <when>
+                        <filecontent pattern="{mapClassname}" />
+                    </when>
+                <perform>
+                    <hint title="Searches for implementation class names via properties have been dropped" effort="1" category-id="mandatory">
+                        <message>Searches for implementation class names through `javax.xml.bind.context.factory` and `javax.xml.bind.JAXBContext` properties have been dropped in Jakarta XML Binding 4.0 for EAP8.
+                            Use of the removed JAXBContext implementation discovery mechanisms should be replaced by one of the supported mechanisms discussed in the links.</message>
+                        <link title="Jakarta XML Binding 4.0 specification"
+                              href="https://jakarta.ee/specifications/xml-binding/4.0/"/>
+                        <link title="Implementation discovery in Jakarta XML Binding 4.0"
+                              href="https://jakarta.ee/specifications/xml-binding/4.0/jakarta-xml-binding-spec-4.0.html#implementation-discovery"/>
+                        <tag>eap8</tag>
+                    </hint>
+                </perform>
+                </iteration>
+            </perform>
+            <where param="mapClassname">
+                <matches pattern="javax.xml.bind.context.factory|javax.xml.bind.JAXBContext" />
+            </where>
+        </rule>
+        <rule id="eap8-xml-binding-00006">
+            <when>
+                <or>
+                <javaclass references="javax.xml.bind.Validator">
+                    <location>IMPORT</location>
+                </javaclass>
+                <javaclass references="javax.xml.bind.JAXBContext.createValidator()">
+                     <location>METHOD_CALL</location>
+                </javaclass>
+                </or>
+            </when>
+            <perform>
+                <hint title="Usage of this Validator interface or retrieval method should be removed" effort="1" category-id="mandatory">
+                    <message>Usage of this Validator interface or retrieval method should be removed.
+                        Passing a schema to `jakarta.xml.bind.Marshaller.setSchema(Schema)` or  `jakarta.xml.bind.UnMarshaller.setSchema(Schema)` is the replacement.</message>
+                    <link title="Jakarta XML Binding 4.0 specification"
+                          href="https://jakarta.ee/specifications/xml-binding/4.0/"/>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="eap8-xml-binding-00007">
+            <when>
+                <or>
+                    <javaclass references="javax.xml.bind.Marshaller.setAdapter({*})">
```

Comment:
I think we're missing `getAdapter()` somewhere around here

Reply:
This one is all to do with a change in the generics accepted as parameters in these methods. Looking at the Marshaller api docs,  getAdapter accepts a parameter  (Class<A> type)  so you may well be right. However it wasn't mentioned in the spec document!

Reply:
I've asked the question on the JIRA